library/public/image.h

```diff
@@ -27,33 +28,72 @@ class F3D_EXPORT image
     }
   };
 
-  image();
+  /**
+   * Create an image from file, the following formats are supported:
+   * PNG, PNM, TIFF, BMP, HDR, JPEG, GESigna, MetaImage, TGA.
+   * Can throw an image::exception in case of failure.
+   */
   explicit image(const std::string& path);
 
+  //@{
+  /**
+   * Default/copy/move constructors/operators
+   */
+  image();
   ~image();
-
   image(const image& img);
   image& operator=(const image& img);
-
   image(image&& img);
   image& operator=(image&& img);
+  //@}
+
+  //@{
+  /**
+   * Comparison operators, uses image::compare
+   */
+  bool operator==(const image& reference) const;
+  bool operator!=(const image& reference) const;
+  //@}
 
+  //@{
+  /**
+   * Set/Get image resolution
+   */
   unsigned int getWidth() const;
   unsigned int getHeight() const;
-
   image& setResolution(unsigned int width, unsigned int height);
+  //@}
 
+  //@{
+  /**
+   * Set/Get image channel count
+   */
   unsigned int getChannelCount() const;
   image& setChannelCount(unsigned int dim);
-
+  //@}
+
+  //@{
+  /**
+   * Set/Get image buffer data
+   * Its size is expected to be width * height * channelCount
+   *
+   */
   image& setData(unsigned char* buffer);
   unsigned char* getData() const;
-
-  bool compare(const image& reference, image& result, double threshold, double& error) const;
-
-  bool operator==(const image& reference) const;
-  bool operator!=(const image& reference) const;
-
+  //@}
+
+  /**
+   * Compare current image to a reference using the provided threshold.
```

Comment:
Can we give the user a hint of what exactly is the threshold? Personally, I have no idea.

Reply:
The VTK doc is not super clear. This is basically a pixel to pixel comparison with averaging and shifting. The value is 256 based but accumulate over multiple pixels.